tests/ignite/metrics/test_confusion_matrix.py

```diff
@@ -55,145 +55,97 @@ def test_multiclass_wrong_inputs():
 
 def test_multiclass_input_N():
     # Multiclass input data of shape (N, )
-    def _test_N():
-        num_classes = 4
-        cm = ConfusionMatrix(num_classes=num_classes)
-        y_pred = torch.rand(10, num_classes)
-        y = torch.randint(0, num_classes, size=(10,)).long()
-        cm.update((y_pred, y))
-        np_y_pred = y_pred.numpy().argmax(axis=1).ravel()
-        np_y = y.numpy().ravel()
-        assert np.all(confusion_matrix(np_y, np_y_pred, labels=list(range(num_classes))) == cm.compute().numpy())
+    def _test_N(y_pred, y, num_classes, batch_size):
 
-        num_classes = 10
         cm = ConfusionMatrix(num_classes=num_classes)
-        y_pred = torch.rand(4, num_classes)
-        y = torch.randint(0, num_classes, size=(4,)).long()
-        cm.update((y_pred, y))
-        np_y_pred = y_pred.numpy().argmax(axis=1).ravel()
-        np_y = y.numpy().ravel()
-        assert np.all(confusion_matrix(np_y, np_y_pred, labels=list(range(num_classes))) == cm.compute().numpy())
+        if batch_size == 1:
```

Comment:
Why do we need this check? we can update then check the `batch size`, right?

Reply:
From execution point of view yes, it's doable. I am just keeping it to make the test case more readable

Reply:
It can work both ways, @vfdev-5 can we hear from you here?